src/backend/env/post.rs

```diff
@@ -133,6 +143,26 @@ impl Post {
         }
     }
 
+    /// Returns the post with some meta information needed by the UI.
+    pub fn with_meta<'a>(&'a self, state: &'a State) -> (&'_ Self, Meta<'_>) {
```

Comment:
Do you know how much it is going to increase the size of a Post. I wonder if we are far away from the 3MB max response size limit in all queries that return posts.

Reply:
No, actually we never ever serve more than 30 posts because everything is paged. Where we bump into limits is when we _search_ for posts to fill a page. So these meta augmentation will be only ever called on 30 posts in the same query.